### PR TITLE
MM-49894: First version of event registry

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

First version of event registry. Iterates over all Rudderstack [tracks](https://www.rudderstack.com/docs/destinations/warehouse-destinations/warehouse-schema/#track) table and joins them across sources. 

Please note the following design choices:
- Ignoring all segment originating events for now.
- Each `tracks` table has its own `base` model. This allows more finegrained control on table level.  A side benefit of this is that new `tracks` can be added without requiring complex incremental logic. This also helped greatly in optimizing execution time.
- Base tables are updated incrementally to reduce resource usage.
- Repeatable code has been extracted to a macro. This will make it easier. 
- There were some missing table definitions for `tracks` tables in sources that needed to be added.

### CODE REVIEW

File: transform/snowflake-dbt/models/event_registry/event_registry.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      - name: id
+        description: A unique id for each event/date/source combination.
+        tests:
+          - unique
+          - not_null
+      - name: date_received_at
+        description: The date that events of the current type were received.
+        tests:
+          - not_null
+      - name: event
```
Comment (catalintomai) [inline transform/snowflake-dbt/models/event_registry/event_registry.yml:19]: @ifoukarakis , maybe just call it `event_table` and the one below `event` or `event_name`? `event_text` seems more appropriate for some event description field.

Comment (ifoukarakis) [reply transform/snowflake-dbt/models/event_registry/event_registry.yml:19]: These are actually columns of the base tables. I reused the same descriptions for the columns as in https://www.rudderstack.com/docs/destinations/warehouse-destinations/warehouse-schema/#table-tracks. Am I missing something here?

We can also discuss names for `event_registry` and `daily_event_stats` tables. I thought of using `events`, but such table exists in other schemas, so I thought of avoiding conflicting names.

Comment (catalintomai) [reply transform/snowflake-dbt/models/event_registry/event_registry.yml:19]: Yeah, not the best (initial) naming imo, but better be consistent in this case I guess.

Comment (ifoukarakis) [reply transform/snowflake-dbt/models/event_registry/event_registry.yml:19]: @catalintomai what about the following:
- Rename `event_registry` to `event_fact`.
- Rename column `event` to `event_table` or `table`.
- Rename `event_text` to `event_name` to `name`.

Alternatively, we can use the naming above but with removing the `event_` prefix. 

`daily_event_stats` should keep the same name, but follow the same conventions.

**Review by catalintomai — APPROVED** [review]

File: transform/snowflake-dbt/models/event_registry/event_registry.yml (hunk 1 above)
Comment (catalintomai) [reply transform/snowflake-dbt/models/event_registry/event_registry.yml:19]: Thanks @ifoukarakis , that makes more sense/easier to follow to me (prefer `event_*` to `*` ).

**Pull request merged**